plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/index.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * External dependencies
+ */
+import { recordEvent } from '@woocommerce/tracks';
+
+const actionButtons = document.querySelectorAll( '.row-actions span' );
+const bulkActions = document.querySelector( '#bulk-action-selector-top' );
+const bulkActionsButton = document.querySelector( '#doaction' );
+const featuredButtons = document.querySelectorAll( '#the-list .featured a' );
+const filterButton = document.querySelector( '#post-query-submit' );
+const productCategory = document.querySelector( '#product_cat' );
+const productType = document.querySelector( '#dropdown_product_type' );
+const searchButton = document.querySelector( '#search-submit' );
+const searchInput = document.querySelector( '#post-search-input' );
+const sortableColumnHeaders = document.querySelectorAll(
+	'.wp-list-table.posts thead .sortable a, .wp-list-table.posts thead .sorted a'
+);
+const stockStatus = document.querySelector( '[name="stock_status"]' );
+
+filterButton.addEventListener( 'click', function () {
+	recordEvent( 'products_list_filter_click', {
+		search_string_length: searchInput.value.length,
+		filter_category: productCategory.value !== '',
+		filter_product_type: productType.value,
+		filter_stock_status: stockStatus.value,
+	} );
+} );
+
+bulkActionsButton.addEventListener( 'click', function () {
```

Comment:
I get an error in my console here when the product list is empty (in my case I filtered by variable products of which I didn't have any).
<img width="1318" alt="Screen Shot 2022-05-16 at 1 58 13 PM" src="https://user-images.githubusercontent.com/2240960/168645145-70349d47-88b4-4eea-bd5d-a15128bfca88.png">

Reply:
Added a check for all elements before binding listeners since it's possible these could be removed/modified by plugins as well.